File: learning_src/practice/change_file_suffix.cpp
#include <iostream>
#include <string>
#include <filesystem>

namespace fs = std::filesystem;

void change_file_suffix(const fs::path& path) {
    for (const auto& entry : fs::recursive_directory_iterator(path)) {
        if (entry.is_regular_file()) {
            const fs::path& filename = entry.path().filename();
            if (filename.extension() == ".cc") {
                fs::path new_filename = filename.stem();
                new_filename += ".c";
                fs::path new_path = entry.path().parent_path() / new_filename;
                fs::rename(entry.path(), new_path);
                std::cout << "Renamed: " << entry.path() << " -> " << new_path << std::endl;
            }
        }
    }
}

int main() {
    const fs::path root_directory = "."; // Change this to the root directory path
    change_file_suffix(root_directory);
    return 0;
}

File: learning_src/practice/change_file_suffix.py
import os

def change_file_suffix(root_directory):
    for root, _, files in os.walk(root_directory):
        for filename in files:
            if filename.endswith(".cc"):
                old_path = os.path.join(root, filename)
                new_filename = os.path.splitext(filename)[0] + ".c"
                new_path = os.path.join(root, new_filename)
                os.rename(old_path, new_path)
                print(f"Renamed: {old_path} -> {new_path}")

if __name__ == "__main__":
    root_folder = os.path.dirname(os.path.abspath(__file__))
    change_file_suffix(root_folder)
